### PR TITLE
chore: update QTT framework & tests to support improvements on left/outer joins

### PR DESCRIPTION
**Pull request by swist**
### Description 
Trying to fix QTT
Supersedes #7464 and #7561

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-functional-tests/src/test/resources/test-runner/correct/join/output.json [hunk 1]
```diff
@@ -1,13 +1,7 @@
 {
   "outputs": [
-    {"topic": "LEFT_OUTER_JOIN", "key": 0, "value": {"NAME": "zero", "VALUE": 0, "F1": null, "F2": null}, "timestamp": 0},
```
Comment (vcrfxia) [inline ksqldb-functional-tests/src/test/resources/test-runner/correct/join/output.json:3]: @spena can you comment on these changes? The testing tool no longer emits join results for left source inputs that don't have any matches on the left side, even though this is a stream-stream left join. This does not seem correct to me?

Comment (vcrfxia) [reply ksqldb-functional-tests/src/test/resources/test-runner/correct/join/output.json:3]: If you want to play around with this yourself, `KsqlTestingToolTest#shouldRunCorrectTests()` fails without these changes.

Comment (spena) [reply ksqldb-functional-tests/src/test/resources/test-runner/correct/join/output.json:3]: The results are correct. The new stream-stream left join semantics do not emit results that do not have a join until the window + grace period has expired or a join is found on the right topic. The output of this test is emitting only the joined records. There are changes we must do to test the left/outer joins, though. But that's something we can do in a follow-up in order to unblock master.

Comment (vcrfxia) [reply ksqldb-functional-tests/src/test/resources/test-runner/correct/join/output.json:3]: What's the default grace period? Also, what changes are needed in order to begin testing left and outer stream-stream joins again? The fact that we only have test coverage for inner stream-stream joins right now is quite a large test gap. (Agreed that we should merge this PR first to unblock master, though.)

**Comment by vcrfxia** [comment]
We've decided to disable the stream-stream left/outer join enhancements in ksqlDB until the grace period is configurable, so I'm closing this PR in favor of https://github.com/confluentinc/ksql/pull/7569 instead.

**Pull request closed without merging**